lib/filterx/filterx-ref.c

```diff
@@ -195,10 +195,15 @@ _filterx_ref_repr(FilterXObject *s, GString *repr)
 {
   FilterXRef *self = (FilterXRef *) s;
 
-  if (self->value->type->repr)
-    return self->value->type->repr(self->value, repr);
+  return filterx_object_repr(self->value, repr);
+}
 
-  return FALSE;
+static gboolean
+_filterx_ref_str(FilterXObject *s, GString *str)
+{
+  FilterXRef *self = (FilterXRef *) s;
+
+  return filterx_object_str(self->value, str);
```

Comment:
We should call the virtual method directly to avoid the truncation.

Reply:
did both for this and the repr() case as well.